packages/client/src/execution/vmexecution.ts

```diff
@@ -179,21 +242,80 @@ export class VMExecution extends Execution {
       const td = await this.vm.blockchain.getTotalDifficulty(headBlock.header.hash())
       this.config.execCommon.setHardforkBy({ blockNumber: number, td, timestamp })
       this.hardfork = this.config.execCommon.hardfork()
-      this.config.logger.info(`Initializing VM execution hardfork=${this.hardfork}`)
-      if (number === BIGINT_0) {
-        const genesisState =
-          this.chain['_customGenesisState'] ?? getGenesis(Number(this.vm.common.chainId()))
-        if (!genesisState) {
-          throw new Error('genesisState not available')
+
+      if (this.config.execCommon.gteHardfork(Hardfork.Prague)) {
```

Comment:
For the client I am actually not sure, maybe also double-think about it: do we here also want to switch to 6800 EIP checks or stick? Does this have advantages/more degrees of freedom as well? Or do we stick here with Prague?